### DISCUSSION TITLE
ZK Stack - Managing Blob Fees

### DISCUSSION THREAD

**Discussion by MikelinoZk** [zkSync-Community-Hub/zksync-developers#833] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [X] Other (please specify below)

### Rust Version

1.75.0

### Operating System

- [ ] Ubuntu
- [X] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

User - "We are testing an Elastic Chain rollup, which anchors on Holesky, and we got into some crazy priority fees when trying to submit blobs, like 2000eth per batch. And this is very interesting how to manage such a situation, and how to prevent.
Could someone direct us to the relevant documentation?" I assume this would be one of the appropriate docs, https://docs.zksync.io/zksync-protocol/rollup/fee-model but any others I should recommend?

### Expected Behavior

Set max limit or other work arounds such as validium mode with off-chain DA

### Current Behavior

Would pay high blob fees

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It seems you are encountering high priority fees for submitting blobs in an Elastic Chain rollup. To manage and potentially reduce these costs, you can refer to the ZKsync documentation on fee structures and gas pricing. Here are some specific sections that might be helpful:

1. **Fee Structure in ZKsync**: This section provides insights into how fees are structured, including the costs associated with publishing pubdata (blobs) on Ethereum, which might be relevant to your situation with high blob fees.
   - [ZKsync Fee Structure](https://docs.zksync.io/zksync-protocol/rollup/fee-model/fee-structure)

2. **How L2 Gas Price Works**: Understanding the mechanism of L2 gas pricing can also provide insights into how costs are calculated and what might be causing high fees.
   - [How L2 Gas Price Works](https://docs.zksync.io/zksync-protocol/rollup/fee-model/how-l2-gas-price-works)

These documents should help you understand the fee mechanisms better and explore potential solutions like setting maximum limits or using off-chain data availability (Validium mode) to manage costs effectively.

For more in-depth discussions and assistance, you can also visit the ZKsync developers discussions on Github:
- [ZKsync Developers Discussions](https://github.com/zkSync-Community-Hub/zksync-developers/discussions)

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by JackHamer09** [comment]
Hi, @MikelinoZk! Do you believe it's an issue with Holesky network or a bug with ZK Stack? It would also be helpful if you provide more details / reproducible code examples for your issue, so that we can help you debug this!

**Reply by MikelinoZk** [reply]
Thank you - will provide docs above and ask for more info

**Comment by sachanayush47** [comment, marked as answer]
What are the nominator and denominator you have set while creating ecosystem? Possibly, your nominator and denominator are set incorrectly. I had this problem in the past when I set these values incorrectly which results in very high fee around 2300. I fixed it by updating values to 1:1

From [Docs](https://code.zksync.io/tutorials/custom-zk-chain/customizing-your-chain#:~:text=The%20base%20token%20nominator,a%201%3A1%20ratio.)
> The base token nominator and denominator are used to define the relation of the base token price with ETH. For example, if we set the nominator to 20 and the denominator to 1, together the relation is 20/1, this would mean that 20 tokens would be given the equivalent value as 1 ETH for gas. For testing purposes, we'll just use a 1:1 ratio.
